Board.java

```diff
@@ -76,62 +81,44 @@ public boolean isGameOver() {
         return hasWinner() || checkForDraw();
     }
 
-    public ArrayList<String> allWinningCombinations() {
-        ArrayList<String> combinations = new ArrayList<>();
+    public ArrayList<List<String>> allWinningCombinations() {
+        ArrayList<List<String>> combinations = new ArrayList<>();
         combinations.addAll(getAllRows());
         combinations.addAll(getAllColumns());
-        combinations.addAll(getAllDiagonals());
+        combinations.addAll(getRightDiagonal());
+        combinations.addAll(getLeftDiagonal());
```

Comment:
can you not have a getDiagonals() function and within that you combine the Left and Right diagonals List<List<Strings>>?